### PR TITLE
temperature for softmax

### PR DESCRIPTION
**Pull request by Farbum**
Small PR to add temperature scaling in the segmentation softmax

### CODE REVIEW

File: rslearn/train/tasks/segmentation.py [hunk 1]
```diff
@@ -291,19 +291,26 @@ def get_metrics(self) -> MetricCollection:
 class SegmentationHead(Predictor):
     """Head for segmentation task."""
 
-    def __init__(self, weights: list[float] | None = None, dice_loss: bool = False):
+    def __init__(
+        self,
+        weights: list[float] | None = None,
+        dice_loss: bool = False,
+        temperature: float = 1.0,
```
Comment (favyen2) [inline rslearn/train/tasks/segmentation.py]: Add test to verify output with temperature is correctly computed (e.g. try with two temperatures and two input logits make sure higher temperature makes the resulting probabilities less extreme). Also might be worth saying in the docstring that this will only affect the output probabilities, not the loss. Actually is that correct, should it be passed to the loss too, or is it only to change the output visualization?

Comment (Farbum) [reply rslearn/train/tasks/segmentation.py]: Yes agreed, it'd be good to have some tests for the Segmentation head, I'll add one.
As for the temperature affecting the loss, it was flowing into the Dice loss but I changed that so that the loss remains unaffected.

**Review by favyen2 — APPROVED** [review]
thanks for the update! and got it now I missed the dice loss thing but yeah makes sense either way as long as docstring matches up
